### DISCUSSION TITLE
Where to set up MultiChannel Lifeline Associations? Root device or endpoints?

### DISCUSSION THREAD

**Discussion by AlCalzone** [zwave-js/zwave-js#1737] [General]
@hanskroner this one is motivated by https://github.com/zwave-js/node-zwave-js/issues/1721

In zwave-js, all lifeline associations are created on the root device, even if endpoints support Multi Channel Associations.
When receiving an unsolicited report from endpoint 0 by a device that supports the same CC on an endpoint and does not support MCA V3 (because in that case, no Multi Channel Lifelines are configured), we map those reports to endpoint 1.

In the above issue, the device supports MCA V3 on the root device and endpoint 1. However, it sends unsolicited reports from endpoint 0, but multi channel encapsulated (which is not allowed, but zwave-js does not care):
```
09:25:50.161 DRIVER « [Node 020] [REQ] [ApplicationCommand]
                      └─[MultiChannelCCCommandEncapsulation]
                        │ source:      0
                        │ destination: 0
                        └─[ThermostatSetpointCCReport]
                            setpoint type: Heating
                            value:         73 °F
```
Since it supports MCA v3, the report is not mapped to endpoint 1 and thus silently ignored.

Should this be mapped anyways? Or could this wrong report caused by the fact that we're not setting up the association on endpoint 1?

**Comment by Xrlls** [comment]
If setting a multichannel lifeline association on the root device, the controller is implicitly telling the device that the destination support multichannel, and the device will send lifeline reporting from the endpoints multichannel encapsulated. It is not necessary to set lifeline associations on the individual endpoints.

**Reply by AlCalzone** [reply]
Okay then if a multichannel (and multichannel association V3) supporting device only sends report like the one above (source endpoint 0) although it supports Thermostat Setpoint CC on endpoint 1 - is that something the device does wrong?

**Reply by Xrlls** [reply]
ok, so actually it's a can of worms; refer to:
https://www.silabs.com/documents/login/miscellaneous/SDS13782-Z-Wave-Management-Command-Class-Specification.pdf
Section 4.31.6, the lifeline association should be multichannel to the root device, which means that it should be after the MULTI_CHANNEL_ASSOCIATION_REPORT_MARKER and set to end point 0. This _should_ trigger multichannel encapsulation, according to table 21, which applies to multi channel association v3. The situation is a lot less stringent on multichannel association v.2 as endpoint ID 0 was not allowed as a destination at the time (table 18)

**Reply by AlCalzone** [reply]
That's what zwave-js is doing (took me a while to figure out this can of worms) and it works fine for most devices. 

However there are some devices out there like the one in the referenced issue, which
1. do support multi channel association v3
2. have a multi channel association as the lifeline (target endpoint 0 on the controller node)

but still send multi channel encapsulated reports from endpoint 0 instead of forwarding reports from the individual endpoints.
